oss-internship-2020/libtiff/test/raw_decode.cc

```diff
@@ -0,0 +1,199 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <array>
+#include <cstring>
+
+#include "gtest/gtest.h"
+#include "helper.h"
+#include "tiffio.h"  // NOLINT(build/include)
+
+// sapi functions:
+//  TIFFOpen
+//  TIFFClose
+//  TIFFGetField
+//  TIFFSetField
+//  TIFFTileSize
+//  TIFFReadRGBATile
+//  TIFFReadEncodedTile
+
+namespace {
+
+constexpr std::array<uint8_t, 6> kCluster0 = {0, 0, 2, 0, 138, 139};
+constexpr std::array<uint8_t, 6> kCluster64 = {0, 0, 9, 6, 134, 119};
+constexpr std::array<uint8_t, 6> kCluster128 = {44, 40, 63, 59, 230, 95};
+
+int CheckCluster(int cluster, const sapi::v::Array<uint8_t>& buffer,
+                 const std::array<uint8_t, 6>& expected_cluster) {
+  uint8_t* target = buffer.GetData() + cluster * 6;
+
+  bool comp = !(std::memcmp(target, expected_cluster.data(), 6) == 0);
+
+  EXPECT_THAT(comp, IsFalse())
+      << "Cluster " << cluster << " did not match expected results.\n"
```

Comment:
Weird ordering again. Those functions seem to be the same. I assume they are duplicated in the original tests as well?